### PR TITLE
feat: add category form

### PR DESCRIPTION
**Pull request by FranekPoland**

### CODE REVIEW

File: src/components/CategoryForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    const formik = useFormik({
+        initialValues: {
+            title: '',
+            published: true
+        },
+        validationSchema: validationSchema,
+        onSubmit: (values) => {
+        },
+    });
+
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:45]: destructure here const {values, validateForm, errors, getFieldProps, validateForm, touched ...anything u want from formik } = formik;

File: src/components/CategoryForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        .min(2, 'Title should be of minimum 2 characters length')
+        .required('Title is required'),
+    published: yup
+        .boolean()
+});
+
+const CategoryForm = () => {
+    const history = useHistory();
+    const formik = useFormik({
+        initialValues: {
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:37]: You will have here initialValues  
{ "title": "", "description": "" }

File: src/components/CategoryForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                    <Box sx={style} >
+                        <Typography variant="h6" component="h2">
+                            Add new category
+                        </Typography>
+                        <FormControl >
+                            <InputLabel htmlFor="title">TITLE</InputLabel>
+                            <Input 
+                            id="title"
+                            name="title"
+                            value={formik.values.title}
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:90]: you can use  {...props.getFieldProps('title')}
use https://github.com/intive/patronage22_lublin_js_cms/tree/main/src/components/Textfield

File: src/components/CategoryForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                            onChange={formik.handleChange}
+                            error={formik.touched.title && Boolean(formik.errors.title)} />
+                        
+                            {formik.errors.title ? <div>{formik.errors.title}</div> : null}
+                        </FormControl>
+                        <div>
+                            Published: <Switch id="published"
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:97]: remove published add description field {...props.getFieldProps('description')}

File: src/types/routes.ts [hunk 5]
```diff
@@ -14,6 +14,7 @@ export enum ROUTES {
   ADD_PRODUCT = '/add-product',
   ORDERS = '/orders',
   CLIENTS = '/clients',
+  ADDCATEGORY ='/addcategory',
```
Comment (dblach-intive) [inline src/types/routes.ts:17]: ADD_CATEGORY

File: src/components/CategoryForm/index.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                    initialValues={formik.initialValues}
+                    onSubmit={handleSubmit}
+                    >
+                    <Box sx={style} >
+                        <Typography variant="h6" component="h2">
+                            Add new category
+                        </Typography>
+                        <FormControl >
+                            <InputLabel htmlFor="title">TITLE</InputLabel>
+                            <Input
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:87]: Try to use TextfieldWrapper from https://github.com/intive/patronage22_lublin_js_cms/tree/main/src/components/Textfield

File: src/components/CategoryForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    const handleSubmit = () => {
+        formik.validateForm().then((err)=> {
+            if (Object.keys(err).length === 0) { 
+                const url = '/api/categories/addCategory';
+                let payload = {
+                    mode: 'raw',
+                    raw: {
+                       title: formik.values.title,
+                       description: '',
+                       published: formik.values.published
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:59]: remove published

**Review by dblach-intive — CHANGES_REQUESTED** [review]

File: src/components/Router/Router.tsx [hunk 8]
```diff
@@ -9,6 +9,7 @@ const Router = () => {
   return (
       <Switch>
         <Route exact path={ROUTES.AUTH} component={AuthLayout}/>
+        <Route exact path={ROUTES.ADDCATEGORY} component={AuthGuard(Layout)} />
```
Comment (dblach-intive) [inline src/components/Router/Router.tsx:12]: CATEGORY_ADD not ADDCATEGORY

File: src/components/Table/index.tsx [hunk 9]
```diff
@@ -49,11 +52,16 @@ const CustomTable: React.FC<CustomTableProps> = ({
     setPage(0);
   };
 
+  const handleOnClick = () => {
+    history.replace("/addcategory");
```
Comment (dblach-intive) [inline src/components/Table/index.tsx:56]: add-category

**Review by dblach-intive — CHANGES_REQUESTED** [review]

File: src/components/CategoryForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+                    }
+                };
+                addCategory(payload).catch((err) => console.log(err));;
+                history.replace("/dashboard");
+            };
+        })
+       
+    }
+
+    const configTextfieldTitle = {
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:71]: why you did that this way ?

File: src/components/CategoryForm/index.tsx [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+        label: "Title",
+        name: "title",
+        value: formik.values.title,
+        fullWidth: true,
+        onChange: formik.handleChange,
+        error: false
+    }
+
+    const configTextfieldDescription = {
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:80]: why you did that this way ?

File: src/components/CategoryForm/index.tsx [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+        name: "description",
+        value: formik.values.description,
+        fullWidth: true,
+        onChange: formik.handleChange,
+        label:"Description",
+        multiline: true,
+        rows: 4
+    }
+
+    if (formik.touched.title && Boolean(formik.errors.title)) {
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:90]: what is this for ?

File: src/components/Table/index.tsx [hunk 13]
```diff
@@ -49,11 +52,16 @@ const CustomTable: React.FC<CustomTableProps> = ({
...
+  const handleOnClick = () => {
+    history.replace("/add-category");
+  }
+
   return (
     <TableContainer component={Paper} sx={{ margin: '32px 0' }}>
       <Typography variant="h4" p={2} sx={{ color: '#0f0f0f' }}>
         Products
       </Typography>
+      <Button variant="contained" style={{float:'right'}} onClick={()=> handleOnClick()}>Add Category</Button>
```
Comment (dblach-intive) [inline src/components/Table/index.tsx:64]: why did u add button here ? add this in dashboard like products button

**Review by dblach-intive — CHANGES_REQUESTED** [review]

File: src/components/CategoryForm/index.tsx [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+        fullWidth: true,
+        onChange: formik.handleChange,
+        error: false
+    }
+
+    const configTextfieldDescription = {
+        name: "description",
+        value: formik.values.description,
+        fullWidth: true,
+        onChange: formik.handleChange,
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:84]: use getFieldProps

File: src/components/CategoryForm/index.tsx [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+        })
+       
+    }
+
+    const configTextfieldTitle = {
+        label: "Title",
+        name: "title",
+        value: formik.values.title,
+        fullWidth: true,
+        onChange: formik.handleChange,
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:76]: use getFieldProps

File: src/components/CategoryForm/index.tsx (hunk 2 above)
Comment (FranekPoland) [reply src/components/CategoryForm/index.tsx:34]: image

You mean something like this?

File: src/components/CategoryForm/index.tsx [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+    const validationSchema = yup.object({
+        title: yup
+            .string()
+            .min(2, 'Title should be of minimum 2 characters length')
+            .required('Title is required'),
+        description: yup
+            .string()
+    });
+
+    const history = useHistory();
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:39]: hooks should be placed on top

File: src/components/CategoryForm/index.tsx [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+                    Add new category
+                </Typography>
+                <Grid 
+                pt={2}
+                item xs={6}>
+                    <TextField 
+                    fullWidth={true}
+                    id="title"
+                    label="title"
+                    {...getFieldProps('title')}></TextField>
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:72]: close tag `<TextField ...... />`

File: src/components/CategoryForm/index.tsx (hunk 1 above)
Comment (dblach-intive) [reply src/components/CategoryForm/index.tsx:52]: destructurization here

File: src/components/CategoryForm/index.tsx [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+import SaveIcon from '@mui/icons-material/Save';
+import {useFormik} from 'formik';
+import * as yup from 'yup';
+import Button from '@mui/material/Button';
+import { useHistory } from "react-router-dom";
+import TextField from '@mui/material/TextField';
+import Grid from '@mui/material/Grid';
+
+
+const style = {
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:13]: can we add styles into separate css module ?

File: src/components/CategoryForm/index.tsx [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+                    {errors.title ? <div>{errors.title}</div> : null}
+                </Grid>  
+                <Grid 
+                pt={2}
+                item xs={6}>
+                    <TextField
+                    fullWidth={true}
+                    multiline={true}
+                    rows={4}
+                    id="description"
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:73]: do we need id ?

File: src/components/CategoryForm/index.tsx [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+            <Box className={classes.style} >
+                <Typography variant="h6" component="h2">
+                    Add new category
+                </Typography>
+                <Grid 
+                pt={2}
+                item xs={6}>
+                    <TextField 
+                    fullWidth={true}
+                    id="title"
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:61]: do we need id ?

File: src/components/CategoryForm/index.tsx [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import SaveIcon from '@mui/icons-material/Save';
+import {useFormik} from 'formik';
+import * as yup from 'yup';
+import Button from '@mui/material/Button';
+import { useHistory } from "react-router-dom";
+import TextField from '@mui/material/TextField';
+import Grid from '@mui/material/Grid';
+import classes from '../CategoryForm/CategoryForm.module.css';
```
Comment (dblach-intive) [inline src/components/CategoryForm/index.tsx:11]: styles not classes

**Review by dblach-intive — APPROVED** [review]

**Pull request closed without merging**